### PR TITLE
Allow creating the user from the account page

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- visit the account page (if you already have a user then talk to @fabricematrat about getting it removed)
- you should get a notice saying that you don't have a user
- click the button, you should get a form
- fill out the form
- your user should get loaded

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7205/

File: jujugui/static/gui/src/app/components/create-payment-user/create-payment-user.js [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+      this.xhrs.push(xhr);
+    },
+
+    /**
+      Update the state when the billing checkbox changes.
+
+      @method _handleBillingSameChange
+      @param {Object} The change event from the checkbox.
+    */
+    _handleBillingSameChange: function(e) {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/create-payment-user/create-payment-user.js]: s/e/evt/

File: jujugui/static/gui/src/app/components/create-payment-user/create-payment-user.js [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+      this.setState({billingAddressSame: e.currentTarget.checked});
+    },
+
+    /**
+      Update the state when the card checkbox changes.
+
+      @method _handleCardSameChange
+      @param {Object} The change event from the checkbox.
+    */
+    _handleCardSameChange: function(e) {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/create-payment-user/create-payment-user.js]: s/e/evt/

File: jujugui/static/gui/src/app/components/account/payment/payment.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+  juju.components.AccountPayment = React.createClass({
+    displayName: 'AccountPayment',
+
+    propTypes: {
+      acl: React.PropTypes.object.isRequired,
+      addNotification: React.PropTypes.func.isRequired,
+      createPaymentMethod: React.PropTypes.func.isRequired,
+      createToken: React.PropTypes.func.isRequired,
+      createUser: React.PropTypes.func.isRequired,
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/account/payment/payment.js]: At the moment all the payment stuff is under a feature flag. So if the flag is turned off the methods don't exist to pass them down, but we only show these components when they're available.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7216/

**Comment by huwshimi** [comment]
Thanks @fabricematrat @frankban :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**